File: src/main/webapp/js/basic/mapping/data-type-mapping.js
/**
 * 数据类型映射页面JS
 * User: zhaoming
 * Date: 13-4-2
 * Time: 下午5:23
 * To change this template use File | Settings | File Templates.
 */
$(function(){
    loadGrid();
});

/**
 *  结算单列表
 */
var lastIndex;

function loadGrid(){
    $('#ordertable').datagrid({
        width:950,
        height:310,
        url: '/dataTypeMapping/query.json',
        rownumbers : true,
        nowrap : true,
        striped : true,
        singleSelect : true,
        loadMsg:'数据加载中......',
        columns:[[
            {field:'databaseType',title:'Database数据类型',width:250},
            {
                field : 'jdbcType',
                title : 'Jdbc数据类型',
                width : 250,
                editor : {
                    type : 'text',
                    options : {
                        max : 100,
                        required : true
                    }
                }
            },
            {
                field : 'javaType',
                title : 'Java数据类型',
                width : 250,
                editor : {
                    type : 'text',
                    options : {
                        max : 100,
                        required : true
                    }
                }
            }
        ]],
        onClickRow : function(rowIndex, rowData) {
            if (lastIndex != rowIndex) {
                $('#ordertable').datagrid('endEdit', lastIndex).datagrid('beginEdit', rowIndex);
            }
            lastIndex = rowIndex;
        }
    });
}


/**
 *  新增数据类型映射
 */
function update() {
    var rows = $('#ordertable').datagrid('acceptChanges').datagrid('getRows');
    for ( var i = 0; i < rows.length; i++) {
        if (rows[i].jdbcType == "" || rows[i].jdbcType == null || rows[i].javaType == "" || rows[i].javaType == null) {
            Commons.showWarn("列表中不允许出现空值");
            return;
        }
    }
    var params = JSON.stringify(rows);

    $.post(springUrl+"/dataTypeMapping/update.json",
        {
            params : params
        },
        function(data){
            if(data.flag){
                Commons.showSuccess("数据类型映射变更成功");
            }else{
                Commons.showError("数据类型映射变更失败");
            }
        }
    );

}

File: src/main/java/com/mic/generator/beforedomain/basic/mapping/DataTypeMapping.java
package com.mic.generator.beforedomain.basic.mapping;

/**
 * 数据类型映射Domain
 * User: admin
 * Date: 13-6-26
 * Time: 下午3:17
 */
public class DataTypeMapping {

    /**
     * Database数据类型
     */
    private String databaseType;

    /**
     * Jdbc数据类型
     */
    private String jdbcType;

    /**
     * Java数据类型
     */
    private String javaType;

    public String getDatabaseType() {
        return databaseType;
    }

    public void setDatabaseType(String databaseType) {
        this.databaseType = databaseType;
    }

    public String getJdbcType() {
        return jdbcType;
    }

    public void setJdbcType(String jdbcType) {
        this.jdbcType = jdbcType;
    }

    public String getJavaType() {
        return javaType;
    }

    public void setJavaType(String javaType) {
        this.javaType = javaType;
    }
}